kie-dmn-validation/src/test/java/org/kie/dmn/validation/ValidatorTest.java

```diff
@@ -0,0 +1,333 @@
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.kie.dmn.core.DMNInputRuntimeTest;
+import org.kie.dmn.core.api.DMNModel;
+import org.kie.dmn.core.api.DMNRuntime;
+import org.kie.dmn.core.util.DMNRuntimeUtil;
+import org.kie.dmn.feel.model.v1_1.Definitions;
+import org.kie.dmn.validation.Msg;
+import org.kie.dmn.validation.ValidationMsg;
+import org.kie.dmn.validation.DMNValidator;
+
+public class ValidatorTest {
+    
+    private static DMNValidator validator;
+    @BeforeClass
+    public static void init() {
+        validator = DMNValidatorFactory.newValidator();
+    }
+    @AfterClass
+    public static void dispose() {
+        validator.dispose();
+    }
+
+    @Test
+    public void testDryRun() {
+        DMNRuntime runtime = DMNRuntimeUtil.createRuntime( "0001-input-data-string.dmn", DMNInputRuntimeTest.class );
+        DMNModel dmnModel = runtime.getModel( "https://github.com/droolsjbpm/kie-dmn", "0001-input-data-string" );
+        assertThat( dmnModel, notNullValue() );
+
+        Definitions definitions = dmnModel.getDefinitions();
+        assertThat( definitions, notNullValue() );
+        
+        DMNValidatorFactory.newValidator().validateModel(definitions);
+    }
+    
+    private Definitions utilDefinitions(String filename, String modelName) {
+        List<ValidationMsg> validateXML;
+        try {
+            validateXML = validator.validateSchema( new File(this.getClass().getResource(filename).toURI()) );
+            assertThat( "using unit test method utilDefinitions must received a XML valid DMN file", validateXML, IsEmptyCollection.empty() );
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+            fail("Unable for the test suite to locate the file for XML validation.");
+        }
+        
+        DMNRuntime runtime = DMNRuntimeUtil.createRuntime( filename, this.getClass() );
+        DMNModel dmnModel = runtime.getModel( "https://github.com/droolsjbpm/kie-dmn", modelName );
+        assertThat( dmnModel, notNullValue() );
+
+        Definitions definitions = dmnModel.getDefinitions();
+        assertThat( definitions, notNullValue() );
+        return definitions;
+    }
+        
+    @Test
+    public void testInvalidXml() throws URISyntaxException {
+        List<ValidationMsg> validateXML = validator.validateSchema(new File(this.getClass().getResource( "invalidXml.dmn" ).toURI()));
+        assertTrue( !validateXML.isEmpty() );
+        
+        validateXML.forEach(System.err::println);
+    }
+    
+    @Test
+    public void testTYPEREF_NO_NS() throws URISyntaxException {
+        List<ValidationMsg> validateXML = validator.validateSchema(new File(this.getClass().getResource( "TYPEREF_NO_NS.dmn" ).toURI()));
+        assertTrue( !validateXML.isEmpty() );
+        
+        validateXML.forEach(System.err::println);
+    }
+    
+    @Test
+    public void testBKM_MISSING_EXPR() {
```

Comment:
I propose to refactor out the repeated code dups in the test only when the interaction between validator and DMNRuntime is more integrated.